### PR TITLE
TPWDCEE-4739

### PR DESCRIPTION
**Pull request by edra-bumci**

### CODE REVIEW

File: Components/Services/NotificationHandler.php [hunk 1]
```diff
@@ -131,25 +131,31 @@ protected function handleSuccess(
...
-     * @param int     $paymentStatusId
+     * @param Order $order
+     * @param int $paymentStatusId
+     *
+     * @param $transactionType
      *
      * @since 1.0.0
      */
-    private function savePaymentStatus(\sOrder $shopwareOrder, Order $order, $paymentStatusId)
+    private function savePaymentStatus(\sOrder $shopwareOrder, Order $order, $paymentStatusId, $transactionType)
```
Comment (ozada) [inline Components/Services/NotificationHandler.php]:
```
if ($transactionType === 'capture-authorization') {
            $sendEmailPaymentId = null;
}
```
1. Use const unstead of direct values : **capture-authorization** could be stored as TRANSACTION_TYPE_CAPTURE_AUTORIZATION.
2. I think you could rewrite **shouldSendStatusMail** and add this logic in this method.

Benefits:
We have one entry point where deciding if we send any email.

File: Components/Services/NotificationHandler.php [hunk 2]
```diff
@@ -115,7 +115,7 @@ protected function handleSuccess(
         // if we already have an order, we can update the payment status directly
         if ($order) {
             $this->logger->debug("Order {$order->getNumber()} already exists, update payment status $paymentStatusId");
-            $this->savePaymentStatus($shopwareOrder, $order, $paymentStatusId);
+            $this->savePaymentStatus($shopwareOrder, $order, $paymentStatusId, $transactionType);
```
Comment (ozada) [inline Components/Services/NotificationHandler.php]: see below.

File: Components/Services/NotificationHandler.php [hunk 3]
```diff
@@ -158,13 +160,19 @@ private function savePaymentStatus(\sOrder $shopwareOrder, Order $order, $paymen
...
+     *
      * @return bool
      *
      * @since 1.0.0
      */
-    public static function shouldSendStatusMail($paymentStatusId)
+    public static function shouldSendStatusMail($paymentStatusId, $transactionType = null)
     {
-        return in_array($paymentStatusId, [
+        $sendEmailPaymentId = $paymentStatusId;
```
Comment (ozada) [inline Components/Services/NotificationHandler.php]: you could avoid  $sendEmailPaymentId = $paymentStatusId and simplify using following statement:
return paymentStatusId **IN** [PAYMENT_TYPES] AND transactionType **NOT IN** [TRANSACTION_TYPES]

File: Components/Services/NotificationHandler.php [hunk 4]
```diff
@@ -158,16 +160,20 @@ private function savePaymentStatus(\sOrder $shopwareOrder, Order $order, $paymen
...
+     *
      * @return bool
      *
      * @since 1.0.0
      */
-    public static function shouldSendStatusMail($paymentStatusId)
+    public static function shouldSendStatusMail($paymentStatusId, $transactionType = null)
     {
-        return in_array($paymentStatusId, [
+        $paymentStatuses = [
```
[the remark this answers is not in the document]
Comment (edra-bumci) [reply Components/Services/NotificationHandler.php]: @ozada class Status is actually not in the extension code but in the shopware source code so I cannot really extract the paymentStatuses array there.

**Pull request merged**